t/t-clone.sh

```diff
@@ -227,16 +263,14 @@ begin_test "clone ClientCert"
       [ $(wc -c < "file2.dat") -eq 75 ]
       [ $(wc -c < "file3.dat") -eq 30 ]
       assert_hooks "$(dot_git_dir)"
+      [ ! -e "lfs" ]
       assert_clean_status
```

Comment:
Why don't we clone here with `git clone` as a test, too?

Reply:
If I understand the question correctly, I think this PR's changes implement that suggestion.

The old version of this test (prior to this PR's changes) would only [run](https://github.com/git-lfs/git-lfs/blob/f1249933f1e33b758743c6cfdf02e473f2935b07/t/t-clone.sh#L236-L238) a single `git clone` after exiting its loop, which is why it [reset](https://github.com/git-lfs/git-lfs/blob/f1249933f1e33b758743c6cfdf02e473f2935b07/t/t-clone.sh#L234) the `http.<url>.sslKey` configuration option, since the second pass through the loop would have [changed](https://github.com/git-lfs/git-lfs/blob/f1249933f1e33b758743c6cfdf02e473f2935b07/t/t-clone.sh#L211) that to refer to the encrypted private key file instead of the unencrypted one.

But this PR's commit 3079fd783c92a2368dd4bd400e6e3606e7ed2f72 moves the `git clone` into the loop, so it now runs twice, once with each version of the private key file.  Quoting from that commit's description:

> As we expect to make several other adjustments to this test in subsequent commits in this PR, we first bring the `git clone` command into the same loop with the `git lfs clone` command, which means we no longer need to reset the `http.<url>.sslKey` configuration option, as we want to use the same value for both cloning commands.  This change also ensures we perform all the same checks with both types of TLS/SSL certificate keys.

Reply:
Sorry about the comment. That didn't made sense.